dronegen/common.go

```diff
@@ -10,37 +10,102 @@ var (
 		Branch: triggerRef{Include: []string{"master", "branch/*"}},
 		Repo:   triggerRef{Include: []string{"gravitational/*"}},
 	}
-
-	volumeTmpfs = volume{
-		Name: "tmpfs",
-		Temp: &volumeTemp{Medium: "memory"},
+	triggerTag = trigger{
+		Event: triggerRef{Include: []string{"tag"}},
+		Ref:   triggerRef{Include: []string{"refs/tags/v*"}},
+		Repo:  triggerRef{Include: []string{"gravitational/*"}},
 	}
+
 	volumeRefTmpfs = volumeRef{
 		Name: "tmpfs",
 		Path: "/tmpfs",
 	}
+
+	// TODO(gus): Set this from `make -C build.assets print-runtime-version` or similar rather
+	// than hardcoding it. Also remove the usage of RUNTIME as a pipeline-level environment variable
+	// (as support for these varies among Drone runners) and only set it for steps that need it.
+	goRuntime = value{raw: "go1.15.5"}
 )
 
-// buildCheckoutCommands builds a list of commands for Drone to check out a git commit
-func buildCheckoutCommands(fips bool) []string {
-	commands := []string{
-		`mkdir -p /go/src/github.com/gravitational/teleport /go/cache`,
-		`cd /go/src/github.com/gravitational/teleport`,
-		`git init && git remote add origin ${DRONE_REMOTE_URL}`,
-		`git fetch origin`,
-		`git checkout -qf ${DRONE_COMMIT_SHA}`,
-		// this is allowed to fail because pre-4.3 Teleport versions don't use the webassets submodule
-		`git submodule update --init webassets || true`,
-		`mkdir -m 0700 /root/.ssh && echo "$GITHUB_PRIVATE_KEY" > /root/.ssh/id_rsa && chmod 600 /root/.ssh/id_rsa`,
-		`ssh-keyscan -H github.com > /root/.ssh/known_hosts 2>/dev/null && chmod 600 /root/.ssh/known_hosts`,
-		`git submodule update --init e`,
-		// do a recursive submodule checkout to get both webassets and webassets/e
-		// this is allowed to fail because pre-4.3 Teleport versions don't use the webassets submodule
-		`git submodule update --init --recursive webassets || true`,
-		`rm -f /root/.ssh/id_rsa`,
-	}
-	if fips {
-		commands = append(commands, `if [[ "${DRONE_TAG}" != "" ]]; then echo "${DRONE_TAG##v}" > /go/.version.txt; else egrep ^VERSION Makefile | cut -d= -f2 > /go/.version.txt; fi; cat /go/.version.txt`)
-	}
-	return commands
+type buildType struct {
+	os      string
+	arch    string
+	fips    bool
+	centos6 bool
+}
+
+type extraVolumes struct {
+	tmpfs          bool
+	tmpDind        bool
+	tmpIntegration bool
+}
+
+// dockerService generates a docker:dind service and includes any extra volumes configured
+// in extraVolumes.
+func dockerService(v extraVolumes) service {
```

Comment:
Change this to 
```go
func dockerService(extraVolumes ...volumeRef) service {
	return service{
		Name:    "Start Docker",
		Image:   "docker:dind",
		Volumes: append(extraVolumes, volumeRefDockersock),
	}
}
```
to make this a bit more flexible.
Then you can drop the `extraVolumes` struct and call this as:
`docerService(volumeRefTmpfs, volumeRefDind)`